### DISCUSSION TITLE
DeDRM_tools 10.0.3: Which versions of KindleForPC, Adobe Digital Editions, KFX Input and Calibre will work with it?

### DISCUSSION THREAD

**Discussion by mQn9x9ne** [noDRM/DeDRM_tools#415] [Q&A]
Hi,

Firstly, I wish to thank the developers of these amazing DeDRM tools and especially ElleKayEm for taking the time to answer the questions in this sub-forum.

Secondly, I have read the post titled [Kindle for Mac, Calibre & noDRM/DeDRM version recommendations #360](https://github.com/noDRM/DeDRM_tools/discussions/360) and felt that I need more up-to-date answers; hence this post.

(1) What versions of the following software are able to remove DRM from an ebook that I am thinking of buying on Amazon?

DeDRM_tools version 10.0.3
KindleForPC-installer-2.0.70350.exe
Calibre 64bit version 6.29.0
KFX Input.zip (which version??)

(2) At the time of writing this post, the current version of Adobe Digital Editions is 4.5.12. Will it work with the following software?

DeDRM_tools version 10.0.3
Calibre 64bit version 6.29.0

Thanks for your answers.

**Comment by ElleKayEm** [comment]
1) Many but not all Kindle books can be decrypted with:
DeDRM 10.0.9
If the book was published this year, you won't be able to download it with older versions of Kindle for PC.
Usually best to keep up-to-date with calibre if possible-- currently 6.29.
Same with KFX Input -- currently 2.5.
Avoid books which say they can only be downloaded with the latest app versions on their product page.  Those will have Amazon's strongest DRM which cannot be removed at this time.

2) Yes, those will work, but recommend DeDRM 10.0.9 for that too.

https://github.com/noDRM/DeDRM_tools/releases/tag/v10.0.9

**Comment by mQn9x9ne** [comment]
Thanks for your reply.

>        If the book was published this year, you won't be able to download it with older versions of Kindle for PC.

I'm terribly disappointed to hear such sad news.

Yes, the Kindle book that I'm thinking of buying was published in February 2023.

In the past, I used to use the following tools to download and deDRM Kindle books (my previous post is [ Unable to remove DRM from Kindle book #967 ](https://github.com/apprenticeharper/DeDRM_tools/issues/967))

```
KindleForPC-installer-1.17.44170.exe
Calibre version 4.5
DeDRM_tools_6.6.3.zip
KFX Input.zip version 1.26.0
```

**Reply by ElleKayEm** [reply]
DeDRM 10.0.9 can still handle many (but again not all) books downloaded with the current version of Kindle for PC.  Exceptions include textbooks and some other publishers which Amazon provides stronger DRM for.  Again look for mention on the product page that it *requires* the latest versions of apps.  Those are the ones to avoid.  But if you can get it elsewhere with Adobe DRM, that may be a better idea.

**Comment by mbinkhorst** [comment]
Maybe this can help you: https://github.com/yihong0618/Kindle_download_helper/

**Comment by mQn9x9ne** [comment]
>DeDRM 10.0.9 can still handle many (but again not all) books downloaded with the current version of Kindle for PC. 

Thanks for your tip.

_Do I need to install KFX Input or run a batch script to prevent Kindle books with file extension *.kfx from being downloaded?_

> Again look for mention on the product page that it requires the latest versions of apps. Those are the ones to avoid.

On the product page of the Kindle book that I'm thinking of buying, there is no mention at all that it requires the latest version of Kindle-For-PC.

Below is a snippet of the product details:

```
Publication date ‏ : ‎ February 13, 2023
Language ‏ : ‎ English
File size ‏ : ‎ 7986 KB
Simultaneous device usage ‏ : ‎ Unlimited
Text-to-Speech ‏ : ‎ Enabled
Screen Reader ‏ : ‎ Supported
Enhanced typesetting ‏ : ‎ Enabled
X-Ray ‏ : ‎ Not Enabled
Word Wise ‏ : ‎ Enabled
Sticky notes ‏ : ‎ On Kindle Scribe
Print length ‏ : ‎ 392 pages 
```

>But if you can get it elsewhere with Adobe DRM, that may be a better idea.

I totally agree with you. However, the Kindle book that I am interested in is unavailable in epub format.

_On the topic of Adobe DRM, can DeDRM_tools 10.0.3 remove DRMs from current Adobe ebooks?_

**Reply by ElleKayEm** [reply]
DeDRM 10.0.9 can handle most KFX format.  You just need the KFX Input plugin to grab all the neccessary files.  There's no known way to stop later versions of Kindle for PC from downloading KFX.  (The main book file will still have an AZW extension.)

> Simultaneous device usage ‏ : ‎ Unlimited

That tells me that if you could use one of the workarounds to get the book in non-kfx format, it would be DRM-free.  But kfx format always has drm.  You probably can remove the DRM in this case though.  Or it might even be a book which doesn't come in kfx -- most do though.  If you don't mind sharing the ASIN, I can see if I notice anything that raises a red flag.

Yes, DeDRM 10.0.3 can handle the newer Adobe DRM.  Recommend DeDRM 10.0.9 which has some bug fixes for retrieving the Adobe key.

**Comment by mQn9x9ne** [comment]
> Maybe this can help you: https://github.com/yihong0618/Kindle_download_helper/

Thanks for the link.

I need a bit more time to translate the page to English first and then try to understand what the content of the web page says.

_A question: do the instructions in that web page show me how to remove DRM from Kindle books published in this year?_ I asked because the Kindle book that I am interested in was published in February 2023.

**Reply by mbinkhorst** [reply]
This is only to download kindle books, the resulting file you can then try to use with DeDRM. If you have little to no experience with Python or using Python scripts it will not be a good solution for you.

**Comment by mQn9x9ne** [comment]
> If you don't mind sharing the ASIN, I can see if I notice anything that raises a red flag.

No, I don't mind. 

Here's the ASIN of the Kindle book: B0BVR2Q1GS

>There's no known way to stop later versions of Kindle for PC from downloading KFX. (The main book file will still have an AZW extension.)

I downloaded a batch script from a forum many years ago. It is used to disable KindleForPC-installer-1.17.44170.exe from downloading Kindle books with *.kfx extension.

Below are the contents of said script:

```
@echo off
set KINDLE=%LocalAppData%\Amazon\Kindle
if not exist "%KINDLE%\application\Kindle.exe" goto :nokindle
if exist "%KINDLE%\storage" if not exist "%KINDLE%\storage\" del /Q "%KINDLE%\storage"
if exist "%KINDLE%\updates" rmdir /S /Q "%KINDLE%\updates"
echo This file disables Kindle for PC downloads. > "%KINDLE%\updates"
echo Kindle for PC downloads are now disabled
goto :exit
:nokindle
echo Cannot disable downloads - Kindle for PC is not installed at expected location
:exit
pause
```

_I wonder if it is still able to be used on the current version of Kindle-For-PC software?_

**Reply by ElleKayEm** [reply]
That script is intended to keep Kindle for PC from updating.  Kindle for PC 1.17 never downloads kfx format.  It was the last version which did not.  There are ways to keep versions up to 1.34 from downloading kfx, but nothing so far after that.

**Reply by ElleKayEm** [reply]
Fairly certain you could DeDRM that book, but I won't guarantee it.

**Comment by mQn9x9ne** [comment]
> This is only to download kindle books, the resulting file you can then try to use with DeDRM.

_Can I use the contents of the web page to download only Kindle books with file extension *.mobi?_ 

>If you have little to no experience with Python or using Python scripts it will not be a good solution for you.

I can learn how to use it on Microsoft Windows OS. I just need to spend more time and effort in learning how to use Python scripts.

**Reply by ElleKayEm** [reply]
The files would be non-kfx...azw3, mobi,...depending on the book...but not kfx.  In other words, DeDRM should handle them.  In the case of the book you were looking at, it would already be DRM-free.

**Comment by mQn9x9ne** [comment]
> That script is intended to keep Kindle for PC from updating. Kindle for PC 1.17 never downloads kfx format. It was the last version which did not. There are ways to keep versions up to 1.34 from downloading kfx, but nothing so far after that.

Thanks for correcting me.

_About said batch script, at what stage should I run it? Is it right after the successful installation of KindleForPC-installer-1.17.44170.exe?_

>Fairly certain you could DeDRM that book, but I won't guarantee it.

That's quite reassuring my friend.

I will let you know if the DeDRM is successful (but before that happens, I wish to learn how to run the Python scripts.)

**Reply by ElleKayEm** [reply]
Yes, you'd want to run that batch file right after installing to avoid it trying to update.  Every time you open it, it will tell you there's a newer version available.  Always remember to choose "skip this version".

**Comment by mQn9x9ne** [comment]
> The files would be non-kfx...azw3, mobi,...depending on the book...but not kfx.

Thanks for your tip.

On the topic of azw3 and mobi.....

_Is there azw4, azw5?_

I know that *.mobi is the earliest format of Kindle books or rather ebooks of file extension *.topaz.

A few years later, in the mid-2000s Kindle books of file extension *.azw3 arrived.

>In the case of the book you were looking at, it would already be DRM-free.

Sorry, you've lost me there.

_I thought all Kindle books come with DRM, yes?_

_Did you mean that if I used the Python scripts to download said Kindle book in *.mobi format, the downloaded *.mobi ebook would be free of DRM, meaning I wouldn't need to apply DeDRM-tools 10.0.3?_

**Reply by ElleKayEm** [reply]
Mobi and topaz are are 2 different older formats not too often seen anymore.  AZW4 is a pdf in an Amazon wrapper.  There was a precursor to KFX format which had an azw6 of higher res images paired with a seperate azw3 file.  AZW3 is also known as KF8.  Most recently KFX which is presumably for Kindle Format 10.

Only KFX always has DRM.  Other formats will be DRM-free if you see "Simultaneous device usage ‏ : ‎ Unlimited" in the details.  So yes if you can use one of the workarounds to get that book in a non-kfx format (usually azw3), you won't need to worry about DRM removal.

I haven't used that "Kindle download helper", but what it says it does is allow you to use the "download & transfer" method from Amazon's website even if you don't have an E-ink Kindle.  There is one other workaround which involves using an older version of Kindle for Android.  Since I do have an E-ink Kindle, I haven't needed to try that either.

**Reply by mQn9x9ne** [reply]
@ElleKayEm

I wish to provide feedback on what I have done with said Kindle book.

I used the latest versions of following tools:

```
DeDRM_tools 10.0.9
Calibre 64bit 6.29.0
KindleForPC-installer-2.0.70350.exe
```

The final result is a Kindle book with file extension azw3

_Question: I forgot to install KFX Input and yet was able to DeDRM the Kindle book.  Did I miss something?_

I remember you wrote last week that 

> You just need the KFX Input plugin to grab all the neccessary files. 

And I forgot to ask you what you meant by "grab all the necessary files".

_Could you tell me what those necessary files are please?_

**Reply by ElleKayEm** [reply]
Looks like that particular book wasn't in kfx format.

When a book is in kfx format, the drm info and some other things are in seperate files instead of being part of the main book file.  The KFX Input plugin grabs them and gathers them into one kfx-zip file.  That file is passed to the DeDRM plugin for decryption.
